lib/rest-api.php

```diff
@@ -87,9 +87,17 @@ function gutenberg_filter_oembed_result( $response, $handler, $request ) {
 				global $wp_embed;
 				$html = $wp_embed->shortcode( array(), $_GET['url'] );
 				if ( $html ) {
+					global $wp_scripts;
+					// Check if any scripts were enqueued by the shortcode, and
+					// include them in the response.
+					$enqueued_scripts = array();
+					foreach ( $wp_scripts->queue as $script ) {
```

Comment:
In my local tests, I'm still having problems when I try to embed http://pinterest.co.uk/heald0081/hair-ideas. I checked that $wp_scripts->queue is empty which I assume was unexpected.
I changed Gutenberg in a poopy.life sandbox to contain:		
```
if ( $html ) {
					global $wp_scripts;
					var_dump( $wp_scripts->queue );
					exit;
```

And I checked $wp_scripts->queue was also empty there. I'm not certain about the reason for this. Maybe I'm doing something wrong.

Reply:
My bet is you tested to make sure it was broken, applied the patch, then tested again? Clear all your embed transients and see if it works :)

Reply:
If not (it hit me a few times!) Can you check that the call to enqueued script happens in the short code?

Reply:
I'm still having problems with the embed of Pinterest, I used a plugin the clear all the transients and the situation persisted. I checked that Pinterest is not a WordPress supported embed in the classic editor https://codex.wordpress.org/Embeds, but JetPack adds a shortcode for it. Do you have JetPack installed in your testing environment? maybe that explains the reason for the difference in the tests because I'm not using JetPack on the environment I'm testing.

Reply:
Yes, this needs Jetpack with the shortcodes module active.

Reply:
Ah sorry for the troubles, it works great with JetPAck shortcodes module active.